app/services/provider_authorisation_analysis.rb

```diff
@@ -0,0 +1,120 @@
+class ProviderAuthorisationAnalysis
```

Comment:
This is nice and generic

Reply:
We'll be able to use this for `make_decisions` and `view_diversity_information` too, when https://github.com/DFE-Digital/apply-for-teacher-training/pull/3338 is merged.